module-2/backend/week_two.md

```diff
@@ -26,20 +35,110 @@ end
 
 Now how would you find the owner of the team with an id of 4?
 
+Team.find(4).owner.name
+
 5. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+
+It is a many to many
+
+class Student
+  has_many :courses
+  has_many :teachers, through: :courses
+end
+
+class Course
+  belongs_to :teacher
+  belongs_to :student
+end
+
+class Teacher
+  has_many :courses
+  has_many :students, through: :courses
+end
+
+
+  create_table "students", force: :cascade do |t|
+    t.string "name"
+  end
+
+
+  create_table "teachers", force: :cascade do |t|
+    t.name "name"
+  end
+
+  create_table "courses", force: :cascade do |t|
+       t.belongs_to :student, index: true
+       t.belongs_to :teacher, index: true
+     end
+
 6. Define foreign key, primary key, and schema.
+
+  A Foreign key is the id or the identifier for how the tables are linked. The foreign key refers the the primary key on the associated table. The primary key is the unique or combined attributes that identify the table. Unless specified it is the id.
+
+  not sure what the schema looks like ?
+
+  * add_foreign_key 'books', 'authors'
+
 7. Describe the relationship between a foreign key on one table and a primary key on another table.
+  The foreign key is the primary key of the other table.
 8. What are the parts of an HTTP response?
+  Status- code like 404, 202, etc.
+  Header- Key/value pair
+  Body - html/css
 
 
 ### Optional Questions
 
 1. Name your five favorite ActiveRecord methods (i.e. methods your models inherit from ActiveRecord) and describe what they do.
+.where (some attribute) => it returns an array on objects that match the specified given criteria
+.group( some attribute).count => returns a hash where the key is the attribute and the value is the count
+.find(:id) => returns the object related to that id
+.join(:some table) => creates a joined table saved in memory that you can further query
+.minimum and .maximum(some attribute) => returns the highest or lowest object
+
 2. Name your three favorite ActiveRecord rake tasks and describe what they do.
+rake db:drop - drops the database
+rake db:create - creates the datbase
+rake db: migrate - sets the migrations
+
 3. What two columns does `t.timestamps null: false` create in our database?
+created_at and updated_at
+
 4. In a database that's holding schools and teachers, what will be the relationship between schools and teachers?
+Many to many
```

Comment:
Interesting! Would a teacher have many schools? I can see a case where they would if they taught different places, but generally I think of teachers only belonging to a single school.